UI/qt-display.hpp

```diff
@@ -5,12 +5,16 @@
 
 #define GREY_COLOR_BACKGROUND 0xFF4C4C4C
 
+class OBSProjector;
+
 class OBSQTDisplay : public QWidget {
 	Q_OBJECT
 	Q_PROPERTY(QColor displayBackgroundColor MEMBER backgroundColor READ
 			   GetDisplayBackgroundColor WRITE
 				   SetDisplayBackgroundColor)
 
+	friend class OBSProjector;
```

Comment:
Can you just make the necessary members public instead?

Reply:
I couldnt connect the signal without making this a friend class, is there some way to mark signals as public to enable OBSProjector to connect them?

Reply:
Do you mean the two mouse events? I'm not sure those parent calls were even doing anything useful; they might be there out of good habit/practice. If you want to keep them in case the base functionality changes, you can add public methods to reach them, and call those from the projector instead.
```
void OBSQTDisplay::OnMouseDoubleClick(QMouseEvent *event)
{
	mouseDoubleClickEvent(event);
}

void OBSQTDisplay::OnMousePress(QMouseEvent *event)
{
	mousePressEvent(event);
}
```

Reply:
Never mind the new public functions; you probably want this to match original intent.

```
void OBSProjector::mouseDoubleClickEvent(QMouseEvent *event)
{
	QWidget::mouseDoubleClickEvent(event);
```
```
void OBSProjector::mousePressEvent(QMouseEvent *event)
{
	QWidget::mousePressEvent(event);
```

Reply:
No this is being used to connect the DisplayCreated signal, in window-project.cpp:83

I can try also passing the mouse events through to the QWidget instead of the display directly as well.

Reply:
K, I'll take a look at the signal after work.

Reply:
>No this is being used to connect the DisplayCreated signal, in window-projector.cpp:83

I'm not seeing an issue. addDrawCallback is being called fine as part of DisplayCreated, at least on Windows. DisplayCreated is already public since `signals:` is the same as `public QT_ANNOTATE_ACCESS_SPECIFIER(qt_signal):`, so you shouldn't need to be a friend.

Also connect is a static member of QObject, so you shouldn't need to use `display.connect`, just
`connect(&display, &OBSQTDisplay::DisplayCreated, addDrawCallback);`

Reply:
Yes that worked, I must have had something different when it was complaining initially.